### ISSUE TITLE
Spaceline goes past the screen when showing the full name of buffer

### ISSUE THREAD

**Issue by duckdodger** [TheBB/spaceline#120]
When I use `org-tree-to-indirect-buffer` in an org file, the new buffer is named with the heading of the tree it's showing, for example "notes.org-Hermitian, Anti-Hermitian and Unitary Operators-1". This whole name gets shown in the modeline and as a result, a part of the modeline goes past the screen from my view such as the segments corresponding to line numbers and the position of the cursor. The screenshot below shows what I'm talking about. 

I don't think it's necessary for the full name of the buffer to be shown always, especially when it's so long. Is there any way I can suppress it barring completely disabling the segment? I'd like for it to show something like "notes.org-Hermitian..." and show the other segments after that.
![screenshot_20161117_165206](https://cloud.githubusercontent.com/assets/17142033/20387464/38e095d0-ace6-11e6-9b83-e0426343b9a7.png)

**Comment by TheBB** [comment]
No, because powerline doesn't read `mode-line-buffer-identification`  but you can PR such a change, I'd be happy to accept it.

**Comment by TheBB** [comment]
Just add an option and trim the output of the `buffer-id` segment.

**Comment by duckdodger** [comment]
> you can PR such a change, I'd be happy to accept it.

Well, I don't know how to do it. :D 

> Just add an option and trim the output of the buffer-id segment.

Hmm, I can try this. I'll look into it. Thank you!

**Comment by TheBB** [comment]
https://github.com/TheBB/spaceline/blob/master/spaceline-segments.el#L84-L86

**Comment by duckdodger** [comment]
Well, I'm stumped. I don't understand how it works and what I have to modify in those lines to trim the output. Can you be a bit more clear? I'm not a programmer, and as such, only know to dabble with variables. So what do I have to do?

**Comment by TheBB** [comment]
If you're not a programmer, don't worry about it. I'll do it myself then.

**Comment by deb0ch** [comment]
Hi !

If I may add a little suggestion, often a lot of the information carried by a file's name is at its end.

Maybe we could consider a truncation that keeps the end of the name.

For instance, a name like `longlongfilename-variant1-variant2.ext`

could be truncated like: `...ant1-variant2.ext`, or (my favourite) something like: `longlon...riant2.ext`.

What do you think of it ? :smiley_cat:

**Comment by TheBB** [comment]
Should be fixed now.
